sympy/core/evalf.py

```diff
@@ -1364,7 +1364,9 @@ def evalf(x, prec, options):
 
 
 class EvalfMixin(object):
-    """Mixin class adding evalf capabililty."""
+    """Mixin class adding evalf capabililty.
+       Methods on the EvalfMixin are available on any Expr instance.
```

Comment:
Either make this the same line or have a blank line between the two lines.

Reply:
It would be better to have a note that explains what this class is used for e.g.
```
"""Mixin class adding evalf capability.

The EvalfMixin class provides the .evalf() method for Expr but is implemented as a mixin
so that it can also be used by non-Expr classes e.g. various Set subclasses and Relational.
To make use of this mixin a class should inherit EvalfMixin and implement the _eval_evalf
method. The mixin will then provide the methods .evalf() and .n() and ensure that the N()
function works on instances of the class.
```

Reply:
The idea behind the issue is that if someone cross-references the `evalf` method in the Sphinx docs, it links to the EvalfMixin class, because that is where the `evalf` method is. But it isn't obvious from there that that method is actually on Expr itself. I guess even for Expr itself, users are forced to understand SymPy's class heirarchy just to see that certain methods are available on all expressions.

Reply:
In my opinion, `EvalfMixin` class still relies on `Basic` attributes like `func`, `is_number` what is not always available on some classes.
I see that `Matrix` had overrided the docs.